indexers/account-worker/src/services/dbService.ts

```diff
@@ -0,0 +1,183 @@
+import { Pool } from 'pg';
+import { config } from '../config';
+import { AccountHistoryUpdateData } from '../interfaces';
+
+let pool: Pool;
+
+const connectDb = async (): Promise<void> => {
+  if (pool) {
+    console.log('Database pool already created.');
+    return;
+  }
+  try {
+    pool = new Pool({
+      host: config.dbHost,
+      port: config.dbPort,
+      user: config.dbUser,
+      password: config.dbPassword,
+      database: config.dbName,
+      max: 10,
+      idleTimeoutMillis: 30000,
+      connectionTimeoutMillis: 5000,
+    });
+
+    const client = await pool.connect();
+    console.log('Connected to PostgreSQL successfully.');
+    client.release();
+
+    pool.on('error', (err, client) => {
+      console.error('Unexpected error on idle PostgreSQL client', err, client);
+    });
+
+  } catch (error) {
+    console.error('Failed to connect to PostgreSQL:', error);
+    throw error;
+  }
+}
+
+const disconnectDb = async (): Promise<void> => {
+  if (pool) {
+    const _discounect = await pool.end();
+    console.log('Disconnected from PostgreSQL.');
+  }
+}
+
+/**
+ * Updates multiple AccountHistory records and the accounts table in a single transaction.
+ * Directly updates accounts table to avoid trigger-based deadlocks.
```

Comment:
This is useful information in the current context (changing how account updates are handled), but won't be necessarily useful in the future.